### PR TITLE
New package: antimicro-qt4

### PR DESCRIPTION
**Pull request by Vlad-Romanov**
Sorry about other pull request, I think i have figure it out :D

### CODE REVIEW

File: srcpkgs/antimicro-qt4/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+revision=1
+wrksrc="antimicro-${version}"
+build_style=cmake
+configure_args="-DUSE_SDL_2=ON -DCMAKE_INSTALL_PREFIX=/usr -DWITH_UINPUT=ON -DWITH_XTEST=ON -DAPPDATA=ON"
+hostmakedepends="pkg-config itstool cmake"
+makedepends="qt5-tools-devel libXtst-devel SDL2-devel qt-devel"
+depends="qt libXtst SDL2 shared-mime-info desktop-file-utils"
+short_desc="GUI program used to map keyboard keys and mouse controls to a gamepad"
+maintainer="VladTheImpaler <vladimir-romanov-ii@yandex.com>"
+license="GPL-3"
```
[the remark this answers is not in the document]
Comment (Vlad-Romanov) [reply srcpkgs/antimicro-qt4/template]: This is how original antimicro template is, no?

[the remark this answers is not in the document]
Comment (cr6git) [reply srcpkgs/antimicro-qt4/template]: `GPL-3.0-or-later`

File: srcpkgs/antimicro-qt4/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'antimicro-qt4'
+pkgname=antimicro-qt4
+version=2.23
+revision=1
+wrksrc="antimicro-${version}"
+build_style=cmake
+configure_args="-DUSE_SDL_2=ON -DCMAKE_INSTALL_PREFIX=/usr -DWITH_UINPUT=ON -DWITH_XTEST=ON -DAPPDATA=ON"
```
Comment (cr6git) [inline srcpkgs/antimicro-qt4/template]: You can remove `-DCMAKE_INSTALL_PREFIX=/usr` it's already set by default.

File: srcpkgs/antimicro-qt4/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+hostmakedepends="pkg-config itstool"
+makedepends="qt5-tools-devel libXtst-devel SDL2-devel qt-devel"
```
Comment (cr6git) [inline srcpkgs/antimicro-qt4/template]: Your package is Qt4, you can remove `qt5-tools-devel`

File: srcpkgs/antimicro-qt4/template [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+depends="desktop-file-utils"
+short_desc="GUI program used to map keyboard keys and mouse controls to a gamepad"
```
Comment (cr6git) [inline srcpkgs/antimicro-qt4/template]: Add a mention to Qt4, to distinguish between the two packages `antimicro` / `antimicro-qt4`

File: srcpkgs/antimicro-qt4/template [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+maintainer="VladTheImpaler <vladimir-romanov-ii@yandex.com>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/AntiMicro/antimicro"
+distfiles="https://github.com/AntiMicro/antimicro/archive/${version}.tar.gz"
+checksum=ef309170612da805992f9194f1973bf38a3174a0856856afedab67f9d927a9ef
+
```
Comment (cr6git) [inline srcpkgs/antimicro-qt4/template]: The two packages install the same files! Please add `conflicts="antimicro>=0"`

File: srcpkgs/antimicro-qt4/template [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+if [ -n "$CROSS_BUILD" ]; then
+	hostmakedepends+=" qt5-host-tools qt5-tools-devel"
```
Comment (cr6git) [inline srcpkgs/antimicro-qt4/template]: Same here: your package is Qt4, not Qt5...
````
if [ -n "$CROSS_BUILD" ]; then
	hostmakedepends+=" qt-host-tools qt-qmake"
fi
````

File: srcpkgs/antimicro-qt4/template [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+fi
+
+pre_configure() {
+
+	export QT_SELECT=4
+
```
Comment (cr6git) [inline srcpkgs/antimicro-qt4/template]: Remove unneeded blank lines.

File: srcpkgs/antimicro-qt4/template (hunk 3 above)
Comment (Vlad-Romanov) [reply srcpkgs/antimicro-qt4/template]: ah, missed remove that dependency, thank you cr6.

File: srcpkgs/antimicro-qt4/template (hunk 4 above)
Comment (Vlad-Romanov) [reply srcpkgs/antimicro-qt4/template]: A mention? In short_desc? OK

File: srcpkgs/antimicro-qt4/template [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+pkgname=antimicro-qt4
+version=2.23
+revision=1
+wrksrc="antimicro-${version}"
+build_style=cmake
+configure_args="-DUSE_SDL_2=ON -DWITH_UINPUT=ON -DWITH_XTEST=ON -DAPPDATA=ON"
+hostmakedepends="pkg-config itstool"
+makedepends="libXtst-devel SDL2-devel qt-devel"
+depends="desktop-file-utils"
+short_desc="Qt4 GUI program used to map keyboard keys and mouse controls to a gamepad"
```
Comment (cr6git) [inline srcpkgs/antimicro-qt4/template]: But now `short_desc` is too long!
````
$ xlint srcpkgs/antimicro-qt4/template
srcpkgs/antimicro-qt4/template:11: short_desc should be less than 72 chars
````

Comment (cr6git) [reply srcpkgs/antimicro-qt4/template]: Change to something like:
````
short_desc="Qt4 GUI program used to map keyboard and mouse controls to a gamepad"
````

Comment (abenson) [reply srcpkgs/antimicro-qt4/template]: Or maybe "Map keyboard and mouse controls to a gamepad"

**Comment by abenson** [comment]
You should target qt5, if possible, instead of qt4

**Comment by cr6git** [comment]
@abenson `antimicro` (Qt5) already exists.

**Comment by Vlad-Romanov** [comment]
Where did branch go I make for this? :/

**Comment by Vlad-Romanov** [comment]
Fail to install dependency???? Why travis-ci is messing up? connection problem with server?

**Pull request closed without merging**